packages/frontend/src/pages/EventDetailsPage/EventDetailsPage.tsx

```diff
@@ -0,0 +1,204 @@
+import { useNavigate, useParams } from 'react-router-dom';
+import { getEvent, getEventNameInLanguage, getEventPlaceInLanguage } from 'src/utils/getEvent';
+import { EventStatus, IEvent, Language } from 'src/types';
+import { Toolbar } from './components/Toolbar/Toolbar';
+import translation from 'src/translations/Russian.json';
+import { EVENTS_ROUTE, languagesLong, VOLUNTEERS_ROUTE } from 'src/constants';
+import css from './EventDetailsPage.module.sass';
+import { Dot } from 'src/components';
+import GroupSvg from 'src/assets/group.svg';
+import UserSvg from 'src/assets/user.svg';
+import TranslationSvg from 'src/assets/translation.svg';
+import MapPinSvg from 'src/assets/mapPin.svg';
+import CalendarSvg from 'src/assets/calendar.svg';
+import TimeSvg from 'src/assets/time.svg';
+
+export interface IToolbarActions {
+  title: string;
+  onClick: () => void;
+}
+
+type EventLanguageSpecificData = {
+  description: string;
+  name: string;
+  place: string;
+  type: Language;
+};
+
+export const EventDetailsPage = () => {
+  const { eventId } = useParams<{ eventId: string }>();
+  const [event, setEvent] = useState<IEvent | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [imageLoading, setImageLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const navigate = useNavigate();
+  const buttonBack = { title: translation.return, onClick: () => navigate(EVENTS_ROUTE) };
+  const menuActions: IToolbarActions[] = [
+    { title: translation.edit, onClick: () => console.log('Redirection to edit event page') },
+    { title: translation.closeRecruitment, onClick: () => console.log('Closing recruitment') },
+  ];
+  const languageDataArray: EventLanguageSpecificData[] = event
+    ? Object.values(event.languageSpecificData.data)
+    : [];
+
+  useEffect(() => {
+    const fetchEvent = async () => {
+      if (eventId) {
+        try {
+          const eventData = await getEvent(eventId);
+          if (eventData) {
+            setEvent(eventData);
+          } else {
+            setError('Event not found');
+          }
+        } catch (err) {
+          console.error('Error fetching event:', err);
+        } finally {
+          setLoading(false);
+        }
+      } else {
+        console.log('No eventId provided');
+        setLoading(false);
+      }
+    };
+
+    fetchEvent();
+  }, [eventId]);
+
+  if (loading) return <div>Loading...</div>;
+  if (error) return <div>{error}</div>;
+  return (
+    <div>
+      <Toolbar menuActions={menuActions} buttonBack={buttonBack} />
+
+      {event ? (
+        <div>
+          {imageLoading && <div>Loading Image...</div>}
+          <div className={css.imageContainer}>
+            {!imageLoading && (
+              <div className={css.overlayTop}>
+                {languageDataArray.map(lang => (
+                  <span key={lang.type} className={css.eventLanguages}>
+                    {lang.type == 'russian' ? 'Рус' : ''}
```

Comment:
please change to `languagesShort[lang.type]`

Reply:
Done